### PR TITLE
Docs Update: Update the Top Level Makefile and cross-compile instructions

### PR DESCRIPTION
**Pull request by jeevantelukula**
Updates the cross compile instructions for AM335x and AM437x to avoid the build errors while compiling the binaries for the target. 
Update the docs for packages required by host while building with Top Level Makefile

### CODE REVIEW

File: source/linux/Overview/Top_Level_Makefile.rst [hunk 1]
```diff
@@ -104,13 +104,13 @@ makefile targets.
...
+.. ifconfig:: CONFIG_part_variant not in ('AM62X', 'AM64X', 'AM62PX', 'AM335X', 'AM437X', 'AM65X')
 
     .. code-block:: console
 
         host# sudo apt-get install build-essential autoconf automake bison flex libssl-dev bc u-boot-tools swig
 
-.. ifconfig:: CONFIG_part_variant in ('AM62PX', 'AM62X', 'AM64X')
+.. ifconfig:: CONFIG_part_variant in ('AM62PX', 'AM62X', 'AM64X', 'AM335X', 'AM437X', 'AM65X')
 
     .. code-block:: console
```
Comment (cshilwant) [inline source/linux/Overview/Top_Level_Makefile.rst:116]: Also add `sudo apt-get install libmpc*` which is needed in AM335x & AM437x.

**Review by cshilwant — CHANGES_REQUESTED** [review]

**Review by cshilwant — DISMISSED** [review]

**Review by uditkumarti — CHANGES_REQUESTED** [review]

File: source/linux/Overview/GCC_ToolChain.rst [hunk 2]
```diff
@@ -145,20 +145,40 @@ libraries.
    - **Compile Directly** : Cross-compile the **helloworld.c** file using the
      cross-compile toolchain directly
 
-     .. code-block:: console
+     .. ifconfig:: CONFIG_part_variant in ('AM335X', 'AM437X')
+
+       .. code-block:: console
+
+          host# ${CROSS_COMPILE_64}gcc ${CFLAGS} --sysroot=${SYSROOT_64} helloworld.c -o helloworld
+
```
[the remark this answers is not in the document]
Comment (cshilwant) [reply source/linux/Overview/GCC_ToolChain.rst:153]: @uditkumarti I won't be `CROSS_COMPILE_32` as this variable is associated with k3r5 devkit in the same file, 

https://github.com/TexasInstruments/processor-sdk-doc/blob/860001e6ddfeb4f492da2ab3baae766dae3cd8d1/source/linux/Overview/GCC_ToolChain.rst?plain=1#L17


Even though the variable mentions `CROSS_COMPILE_64` it's documented to point to intended tool-chain, 

https://github.com/TexasInstruments/processor-sdk-doc/blob/860001e6ddfeb4f492da2ab3baae766dae3cd8d1/source/linux/Overview/GCC_ToolChain.rst?plain=1#L14

File: source/linux/Overview/GCC_ToolChain.rst [hunk 3]
```diff
@@ -145,20 +145,40 @@ libraries.
...
+       Pass the CFLAGS: :file:`-march=armv7-a` :file:`-mthumb` :file:`-mfpu=neon` :file:`-mfloat-abi=hard`
```
Comment (StaticRocket) [inline source/linux/Overview/GCC_ToolChain.rst:154]: These aren't files, just use
```suggestion
       Pass the CFLAGS: ``-march=armv7-a -mthumb -mfpu=neon -mfloat-abi=hard``
```

**Review by StaticRocket — CHANGES_REQUESTED** [review]

File: source/linux/Overview/Top_Level_Makefile.rst [hunk 4]
```diff
@@ -104,17 +104,17 @@ makefile targets.
...
 
         host# sudo apt-get install build-essential autoconf automake bison flex libssl-dev bc u-boot-tools swig
 
-.. ifconfig:: CONFIG_part_variant in ('AM62PX', 'AM62X', 'AM64X')
+.. ifconfig:: CONFIG_part_variant in ('AM62PX', 'AM62X', 'AM64X', 'AM335X', 'AM437X', 'AM65X', 'AM57X')
 
     .. code-block:: console
 
-        host# sudo apt-get install build-essential autoconf automake bison flex libssl-dev bc u-boot-tools swig python3 python3-pip
+        host# sudo apt-get install build-essential autoconf automake bison flex libmpc* libssl-dev bc u-boot-tools swig python3 python3-pip
```
Comment (StaticRocket) [inline source/linux/Overview/Top_Level_Makefile.rst:117]: Is there a particular version of libmpc you want to call out here? A glob match is good for short term but long term it makes it difficult to decipher what the actual dependency was.

File: source/linux/Overview/GCC_ToolChain.rst [hunk 5]
```diff
@@ -9,12 +9,22 @@ GCC ToolChain Setup
...
 Before compiling any of the sources referenced in this document, set the cross compiler paths to the toolchains packaged in the Processor SDK [Recommended] as shown below. Refer to :ref:`yocto-toolchain` section for more details on usage.
 
-.. code-block:: console
+.. ifconfig:: CONFIG_part_variant in ('AM335X', 'AM437X')
+
+   .. code-block:: console
+
+      host# export CROSS_COMPILE_64="${SDK_INSTALL_DIR}/linux-devkit/sysroots/x86_64-arago-linux/usr/bin/arm-oe-linux-gnueabi/arm-oe-linux-gnueabi-"
+      host# export SYSROOT_64="${SDK_INSTALL_DIR}/linux-devkit/sysroots/armv7at2hf-neon-oe-linux-gnueabi"
+      host# export CC_64="${CROSS_COMPILE_64}gcc --sysroot=${SYSROOT_64}"
```
Comment (StaticRocket) [inline source/linux/Overview/GCC_ToolChain.rst:18]: As @jeevantelukula indicated:
> For AM335x/AM437x devices, CROSS_COMPILE_64, SYSROOT_64 and ENV_SETUP_64 paths are to be modified with appropritate locations in the installer.

**Review by StaticRocket — CHANGES_REQUESTED** [review]

File: source/linux/Overview/Top_Level_Makefile.rst (hunk 4 above)
Comment (jeevantelukula) [reply source/linux/Overview/Top_Level_Makefile.rst]: Latest version of the packages should be fine. And added all the actual dependencies.

File: source/linux/Overview/GCC_ToolChain.rst (hunk 5 above)
Comment (StaticRocket) [reply source/linux/Overview/GCC_ToolChain.rst:18]: There's also the issue of, none of those toolchains are 64 bit. The arm toolchain section uses the `CROSS_COMPILE_32` variable. Does the devkit do something odd and need the `*_64` variables populated regardless?

**Review by StaticRocket — CHANGES_REQUESTED** [review]

File: source/linux/Overview/GCC_ToolChain.rst (hunk 5 above)
Comment (jeevantelukula) [reply source/linux/Overview/GCC_ToolChain.rst]: Updated the suffixes for those toolchains.

**Comment by StaticRocket** [comment]
AM57 is not mentioned in any of these common document changes. Is that intentional?

**Review by nsaulnier-ti — APPROVED** [review]
Without actually testing the steps, changes look good

File: configs/AM57X/AM57X_linux_config.txt [hunk 6]
```diff
@@ -24,6 +24,7 @@ Replacement Variables
 '__SDK_DOWNLOAD_URL__'           : '`AM57X-Linux-SDK-Download-page <http://software-dl.ti.com/processor-sdk-linux/esd/AM57X/latest/index_FDS.html>`__'
 '__LINUX_UBUNTU_VERSION_LONG__'  : '18.04 (64-bit)'
 '__LINUX_UBUNTU_VERSION_SHORT__' : '18.04'
+'__OPTEE_PLATFORM_FLAVOR__'      : 'am57x'
```
Comment (StaticRocket) [inline configs/AM57X/AM57X_linux_config.txt:27]: Isn't it `am57xx` ?

https://github.com/OP-TEE/optee_os/blob/b8125477b03b73ceeccc65c33e4b8cd93d3fff1a/core/arch/arm/plat-ti/conf.mk#L13

**Review by StaticRocket — DISMISSED** [review]

File: configs/AM57X/AM57X_linux_config.txt (hunk 6 above)
Comment (cshilwant) [reply configs/AM57X/AM57X_linux_config.txt:27]: Yeah, it should be am57xx

@jeevantelukula please plan to update it

**Review by cshilwant — CHANGES_REQUESTED** [review]
Update optee machine for AM57x

**Review by cshilwant — APPROVED** [review]

**Review by StaticRocket — APPROVED** [review]

**Pull request merged**